PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/actions/SparkSubmitJobAction.kt

```diff
@@ -90,4 +106,18 @@ class SparkSubmitJobAction : AzureAnAction() {
             model.isClusterSelectionEnabled = true
         }
     }
-}
\ No newline at end of file
+}
+
+open class SparkSelectTypeAndSubmitJobActionDelegate(private val sparkType: SparkApplicationType) : SparkApplicationTypeOptionAction() {
+    override fun getSparkApplicationType(): SparkApplicationType {
+        return this.sparkType
+    }
+    override fun runPostAction(e: AnActionEvent) {
+        val action = ActionManagerEx.getInstance().getAction("Actions.SubmitSparkApplicationAction")
+        action?.actionPerformed(e)
+    }
+}
+
+class LivySparkSubmitJobAction : SparkSelectTypeAndSubmitJobActionDelegate(SparkApplicationType.HDInsight)
```

Comment:
Can we do it like this:

```suggestion
class LivySparkSelectAndSubmitAction : SparkSelectTypeAndSubmitAction(ActionManagerEx.getInstance().getAction("SetDefaultSparkRunConfigurationTypeHDInsight"))
```